value.go

```diff
@@ -690,7 +690,9 @@ func (self Value) export() interface{} {
 				}
 				value := object.get(name).export()
 				t = reflect.TypeOf(value)
-				if state == 0 {
+				if t == nil {
```

Comment:
This looks like it would prevent common type for nil?

Would this make more sense:
```go
var k reflect.Kind
if t != nil {
	k = t.Kind()
}
if state == 0 {
	kind = k
	state = 1
} else if state == 1 && kind != k {
       	state = 2
}
```